### PR TITLE
feat: curioweb: Improve UX, add top menu

### PR DESCRIPTION
**Pull request by snadrus**
## Related Issues
A Unified menu and some style improvements / consistency. 

Images:
Main Dash:
image

Edit Configuration, Picker:
Screenshot 2024-04-18 at 6 02 52 PM

Edit Configuration, Editor:
Screenshot 2024-04-18 at 6 11 20 PM

Sectors Page:
Screenshot 2024-04-18 at 6 13 47 PM

PoRep:
image

Documentation page not shown as it links to the known docs. 

## Proposed Changes
Removed Clarity (VMWare CSS Layer) except from the Config Picker page. 
Built a unified style like Clarity (font, size, colors)
Hide the page until it's mostly styled (though maybe not loaded). 
Bootstrap 5 (the only one that's module-capable), mostly requesting dark components. 

## Additional Info
The super-small screen makes a hamburger menu that doesn't work. I doubt this is a real problem, but it's ugly. 

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: curiosrc/web/static/pipeline_porep.html [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
     <title>Lotus Provider PoRep Pipeline</title>
     <script src="https://unpkg.com/htmx.org@1.9.5" integrity="sha384-xcuj3WpfgjlKF+FXhSQFQ0ZNr39ln+hwjN3npfM9VBnUskLolQAcN80McRIVOPuO" crossorigin="anonymous"></script>
+    <script type="module" src="/ux/curio-ux.mjs"></script>
     <script type="module" src="chain-connectivity.js"></script>
     <link rel="stylesheet" href="main.css">
     <link rel='stylesheet' href='https://cdn.jsdelivr.net/npm/hack-font@3.3.0/build/web/hack-subset.css'>
 </head>
-<body>
+<body style="visibility: hidden">
+<curio-ux>
```
Comment (magik6k) [inline curiosrc/web/static/pipeline_porep.html:11]: This wrapper seems to break the htmx `hx-get` below, which breaks this page (pipeline state tables don't render at all now)

2024-04-19-095834_3837x912_scrot

Comment (magik6k) [reply curiosrc/web/static/pipeline_porep.html:11]: For reference before the changes here it looked like this
2024-04-19-100542_1564x1342_scrot

File: curiosrc/web/static/chain-connectivity.mjs [hunk 2]
```diff
@@ -60,8 +49,8 @@ window.customElements.define('chain-connectivity', class MyElement extends LitEl
         ${this.data.map(item => html`
         <tr>
             <td>${item.Address}</td>
-            <td>${item.Reachable ? html`<span class="success">ok</span>` : html`<span class="error">FAIL</span>`}</td>
-            <td>${item.SyncState === "ok" ? html`<span class="success">ok</span>` : html`<span class="warning">${item.SyncState}</span>`}</td>
+            <td>${item.Reachable ? html`<span class="alert alert-success">ok</span>` : html`<span class="alert altert-danger">FAIL</span>`}</td>
+            <td>${item.SyncState === "ok" ? html`<span class="alert alert-success">ok</span>` : html`<span class="alert alert-warning">${item.SyncState}</span>`}</td>
```
Comment (magik6k) [inline curiosrc/web/static/chain-connectivity.mjs:53]: Not sure about pulling in the whole boostrap just for those alert classes; They don't seem to fit too well to the rest of the styling

**Review by magik6k — COMMENTED** [review]
Tables could use some restyling to match clarity, or at least the background color + there seem to be some regressions on the sector table pages that should get fixed, but other than that I like where this is going.

(I'll try to start some light refactor to Lit + Jsonrpc to start the train of putting all the ui code within one framework.)

2024-04-19-095711_3827x1862_scrot

**Comment by LexLuthr** [comment]
Apart from what Magik pointed out:
1. Add a grid to main page
2. Docs should open in a new tab
3. We should filter out completed task list for winpost tasks. They clog the display and prevent users to see useful info.

**Comment by magik6k** [comment]
(Force-push undoing commits from #11904 which I've accidentally pushed here)

**Review by LexLuthr — APPROVED** [review]

**Review by magik6k — APPROVED** [review]

File: curiosrc/web/static/chain-connectivity.mjs (hunk 2 above)
Comment (snadrus) [reply curiosrc/web/static/chain-connectivity.mjs:53]: Bootstrap is painting the top menu and soon will paint some other widgets. It's also doing most of the styling for the tables, so I'd rather not remove it until we have fixes for those.

**Pull request merged**